components/certificate-mgt/org.wso2.carbon.identity.certificate.management.core/src/main/java/org/wso2/carbon/identity/certificate/management/core/ApplicationCertificateManagementServiceImpl.java

```diff
@@ -0,0 +1,183 @@
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.certificate.management.core;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.certificate.management.core.constant.CertificateMgtErrors;
+import org.wso2.carbon.identity.certificate.management.core.dao.impl.ApplicationCertificateManagementDAOImpl;
+import org.wso2.carbon.identity.certificate.management.core.exception.CertificateMgtClientException;
+import org.wso2.carbon.identity.certificate.management.core.exception.CertificateMgtException;
+import org.wso2.carbon.identity.certificate.management.core.model.Certificate;
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateMgtExceptionHandler;
+import org.wso2.carbon.identity.certificate.management.core.util.CertificateValidator;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+
+import java.util.UUID;
+
+/**
+ * This class is responsible for managing trusted certificates of applications of a tenant.
+ * @deprecated Use {@link CertificateManagementServiceImpl} instead.
+ */
+public class ApplicationCertificateManagementServiceImpl implements ApplicationCertificateManagementService {
+
+    private static final Log LOG = LogFactory.getLog(ApplicationCertificateManagementServiceImpl.class);
+    private static final ApplicationCertificateManagementService INSTANCE =
+            new ApplicationCertificateManagementServiceImpl();
+    private static final ApplicationCertificateManagementDAOImpl DAO = new ApplicationCertificateManagementDAOImpl();
+    private static final CertificateValidator CERTIFICATE_VALIDATOR = new CertificateValidator();
+
+    private ApplicationCertificateManagementServiceImpl() {
+    }
+
+    public static ApplicationCertificateManagementService getInstance() {
+
+        return INSTANCE;
+    }
+
+    /**
+     * Add a certificate.
+     *
+     * @param certificate  Certificate information.
+     * @param tenantDomain Tenant domain.
+     * @return Auto incremented integer id of the Certificate.
+     * @throws CertificateMgtException If an error occurs while adding the certificate.
+     */
+    @Override
+    @Deprecated
+    public int addCertificate(Certificate certificate, String tenantDomain) throws CertificateMgtException {
+
+        LOG.debug("Adding certificate with name: " + certificate.getName());
```

Comment:
Change other places as well
```suggestion
        if (LOG.isDebugEnabled()) {
            LOG.debug("Adding certificate with name: " + certificate.getName());
        }
```

Reply:
As @malithie mentioned in https://github.com/wso2/carbon-identity-framework/pull/6043#discussion_r1809115417, it is not necessary to add the condition if we are not processing strings considerably like using formatters, builders etc